code/game/objects/effects/spiders.dm

```diff
@@ -168,23 +172,27 @@
 	if(isturf(loc))
 		amount_grown += rand(0,2)
 		if(amount_grown >= 100)
-			if(!grow_as)
-				grow_as = pick(typesof(/mob/living/simple_animal/hostile/poison/giant_spider))
-			var/mob/living/simple_animal/hostile/poison/giant_spider/S = new grow_as(loc)
-			S.faction = faction.Copy()
-			S.master_commander = master_commander
-			if(player_spiders && !selecting_player)
-				selecting_player = 1
-				spawn()
-					var/list/candidates = SSghost_spawns.poll_candidates("Do you want to play as a giant spider?", ROLE_GSPIDER, TRUE, source = S)
-
-					if(candidates.len)
-						var/mob/C = pick(candidates)
-						if(C)
-							S.key = C.key
-							if(S.master_commander)
-								to_chat(S, "<span class='biggerdanger'>You are a spider who is loyal to [S.master_commander], obey [S.master_commander]'s every order and assist [S.master_commander.p_them()] in completing [S.master_commander.p_their()] goals at any cost.</span>")
-			qdel(src)
+			if(SSmobs.giant_spiders <= SPIDER_HARD_CAP)
```

Comment:
You can simplify this diff right down by simply doing
```dm
if(SSmobs.giant_spiders > SPIDER_HARD_CAP)
    return
```

Use early returns where possible, it makes code a lot less painful.